components/p3a/features.h

```diff
@@ -0,0 +1,19 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_P3A_FEATURES_H_
+#define BRAVE_COMPONENTS_P3A_FEATURES_H_
+
+#include "base/feature_list.h"
+
+namespace brave {
+namespace features {
+
+BASE_DECLARE_FEATURE(kSTAR);
```

Comment:
there are quite a lot of changes here and it seems like more of them need to be guarded by the flag in case there are crashes, etc... in the new code. Ideally any non-trivial changes for this would have the feature flag guard to prevent that.

Reply:
Almost all code related to STAR message preparation, epoch rotation and uploading is done in the `BraveP3AMessageManager`.  There are many usages in the class to prevent STAR usage if not enabled